contracts/LockedAccount.sol

```diff
@@ -95,57 +95,71 @@ contract LockedAccount is Ownable, TimeSource, ReturnsErrors, Math {
+                // before burn happens, investor must make allowance to locked account
+                if (neumarkToken.allowance(investor, address(this)) < a.neumarksDue) {
                     return logError(Status.NOT_ENOUGH_NEUMARKS_TO_UNLOCK);
                 }
-                if (!neumarkToken.transferFrom(msg.sender, address(this), a.neumarksDue)) {
+                if (!neumarkToken.transferFrom(investor, address(this), a.neumarksDue)) {
                     return logError(Status.NOT_ENOUGH_NEUMARKS_TO_UNLOCK);
                 }
                 // burn neumarks corresponding to unspent funds
                 neumarkCurve.burnNeumark(a.neumarksDue);
                 // take the penalty if before unlockDate
                 if (currentTime() < a.unlockDate) {
                     uint256 penalty = fraction(a.balance, penaltyFraction);
-                    // allowance for penalty to pool contract
-                    require(ownedToken.approve(address(feePool), penalty));
-                    // add to distribution
-                    feePool.addFee(penalty);
+                    // distribute penalty
+                    if (isContract(penaltyDisbursalAddress)) {
+                        // transfer to contract
+                        require(assetToken.approveAndCall(penaltyDisbursalAddress, penalty, ""));
+                    } else {
+                        // transfer to address
+                        require(assetToken.transfer(penaltyDisbursalAddress, penalty));
+                    }
+                    PenaltyDisbursed(investor, penalty, penaltyDisbursalAddress);
                     a.balance = _subBalance(a.balance, penalty);
                 }
             }
             // transfer amount back to investor - now it can withdraw
-            require(ownedToken.transfer(msg.sender, a.balance));
+            require(assetToken.transfer(investor, a.balance));
             // remove balance, investor and
-            FundsUnlocked(msg.sender, a.balance);
-            _removeInvestor(msg.sender, a.balance);
+            FundsUnlocked(investor, a.balance);
+            _removeInvestor(investor, a.balance);
         }
         return Status.SUCCESS;
     }
 
+    // unlocks msg.sender tokens by making them withdrawable in assetToken
+    // expects number of neumarks that is due to be available to be burned on msg.sender balance - see comments
+    // if used before longstop date, calculates penalty and distributes it as revenue
+    function unlock()
+        onlyStates(LockState.AcceptingUnlocks, LockState.ReleaseAll)
+        public
+        returns (Status)
+    {
+        return unlockFor(msg.sender);
+    }
+
     // this allows to unlock and allow neumarks to be burned in one transaction
     function receiveApproval(address from, uint256 _amount, address _token, bytes _data)
+        onlyStates(LockState.AcceptingUnlocks, LockState.ReleaseAll)
```

Comment:
then you have this: `require(_token == address(neumarkToken));` check below. but a good catch! I have a test for this case as well